### ISSUE TITLE
Issues with tag colors in Tagmanager

### ISSUE THREAD

**Issue by gernert** [Jermolene/TiddlyWiki5#643]
Regarding:
- Tag colors not always shown correctly in Tag manager #596 (closed)
- Add hexadecimal input to TagManager (feature request) #597 (closed)

There are still some issues with the colors (and hexadecimal values) in the Tag manager:

1) The default yellow/brown tag color shows black in the color picker
2) The default yellow/brown tag color has no hexadecimal value
3) Three digit hexadecimal color values show black in the color picker
4) Six digit hexadecimal color values are shown correctly in the color picker

Above checked with TW 5.0.12-beta, Firefox v29.01, and Chrome v35.0.1916.114 m on Windows 7 64-bit.

**Comment by twMat** [comment]
[Still] confirming this

New laptop (and new OS) the problems still persist. Win10, FF & Chrome

![tagmanager](https://cloud.githubusercontent.com/assets/4534056/12638634/182eec80-c59f-11e5-82af-34429265adf0.PNG)

**Comment by tobibeer** [comment]
As for this issue, I think it's safe to say that you can close it, since all these issues are browser issues that leave nothing actionable on the TiddlyWiki end.

I'd encourage you to post a feature request / leave a comment at one such at your favorite browser's repo while pointing here, maybe.

**Comment by twMat** [comment]
>all these issues are browser issues

I didn't realize this. Particularly since the problem (in at least my case) is both in FF and Chrome, and for years and years. Are you sure?

**Comment by tobibeer** [comment]
Not 100% sure, but I'd give it a 95%. ;-)

**Comment by ssokolow** [comment]
While I agree that the browsers' colour pickers *should* accept any valid CSS colour identifier, couldn't TiddlyWiki provide a workaround using `getComputedStyle` to force named colours and three-digit hex colours to be converted to something usable?

**Comment by tobibeer** [comment]
@ssokolow, right, haven't thought of that... sounds like an option worth exploring. I think we all agree that we don't want any dictionaries of named colors.

**Comment by pmario** [comment]
I think, there is a problem. ... The tag manager doesn't initialize the color selector box. So they are all black. And that's what the picker shows. ...

**Comment by ssokolow** [comment]
@pmario 
I can't speak for whatever browser you're using, but the pickers get set up properly for Firefox on any field containing a value of the form `#RRGGBB`. The problem is that, to the best of my knowledge, no other format is required to be supported by the standard and some of the fields contain other things like `#fff` or `rgba(0,200,0,0.7)` or `inherit` or `<< colour foreground>>`.

That's why I suggested using `getComputedStyle` to resolve it to an `rgba(...)` value and then manually parsing and re-serializing that into an `#RRGGBB` format that could be fed into the picker. (Since, even if it would discard information to actually change the value using the picker, the picker button makes a good preview for what the current value resolves to.)
